airbyte-integrations/connectors/source-greenhouse/source_greenhouse/client.py

```diff
@@ -35,44 +35,103 @@
 
 
 def paginator(request, **params):
-    """Split requests in multiple batches and return records as generator"""
+    """
+    Split requests in multiple batches and return records as generator.
+    Use recursion in case of nested streams.
+    """
     rows = request.get(**params)
-    yield from rows
+    nested_names = params.pop("nested_names", None)
+    if nested_names:
+        if len(nested_names) > 1:
+            params["nested_names"] = params["nested_names"][1:]
+        for row in rows:
+            yield from paginator(getattr(request(**params, object_id=row["id"]), nested_names[0]))
+    else:
+        yield from rows
     while request.records_remaining:
         rows = request.get_next()
         yield from rows
 
 
+class HarvestClient(Harvest):
```

Comment:
we should make a ticket to convert this to the CDK

Reply:
We have an issue for that: https://github.com/airbytehq/airbyte/issues/1943. There is also an issue to add the incremental sync support: https://github.com/airbytehq/airbyte/issues/1386. Those 2 issue may be merged into one.

I'll add a comment in the code following that.